agent/CMakeLists.txt

```diff
@@ -11,11 +11,21 @@ option (BUILDSWIGNODE "Build swig node modules." OFF)
 option (BUILDTESTS "Override the addition of tests" OFF)
 add_subdirectory(external/mraa EXCLUDE_FROM_ALL)
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBNM REQUIRED libnm)
+pkg_check_modules(GLIB REQUIRED glib-2.0)
+
 include_directories(${PROJECT_SOURCE_DIR}/external/mraa/api)
 include_directories(${CMAKE_SOURCE_DIR}/agent)
+include_directories(${GLIB_INCLUDE_DIRS})
+include_directories(${LIBNM_INCLUDE_DIRS})
+link_directories(${LIBNM_LIBRARY_DIRS})
```

Comment:
I guess you don't need link_directories since you've already had target_link_libraries below.